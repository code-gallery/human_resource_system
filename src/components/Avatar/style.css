.Avatar:before,
.Avatar:after {
  display: table;
  content: ' ';
}

.Avatar:after {
  clear: both;
}

.Avatar img {
  display: inline;
  border-radius: 100%;
  width: 40px;
  height: 40px;
  border: 2px solid #fff;
}

.Avatar-withLabel img {
  float: left;
  margin: 0 15px 0 0;
}

.Avatar span {
  font-size: 14px;
  display: inline;
  overflow: hidden;
  position: relative;
  top: 5px;
  color: #44474c;
}

.Avatar-blue img {
  border: 2px solid #bfcae3;
}

.Avatar-size-45 img {
  width: 45px;
  height: 45px;
}

.Avatar-size-109 img {
  width: 109px;
  height: 109px;
}

.Avatar .Avatar-icon-danger {
  background-image: url(../Profile/ProfileInfo/assets/icon-danger.png);
  background-repeat: no-repeat;
  background-size: 22px;
  width: 22px;
  height: 22px;
  top: 21px;
  right: 16px;
  display: inline-block;
  margin: 0;
  margin-right: -22px;
}
