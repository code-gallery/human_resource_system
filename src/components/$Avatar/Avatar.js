import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import isString from 'lodash/isString'
import AssignAvatarImg from '../../assets/images/avatar-1.png'
import './style.css'

const DEFAULT_AVATAR = 'https://collaborativecbt.com/wp-content/uploads/2016/12/default-avatar-768x768.png'

const Avatar = ({ shadow, imgUrl, alt, className, status }) => {
  const src = isString(imgUrl) ? imgUrl : status ? AssignAvatarImg : DEFAULT_AVATAR

  return (
    <img
      className={classNames('$Avatar', className, { 'has-shadow': shadow })}
      src={src}
      alt={alt}
    />
  )
}

Avatar.propTypes = {
  shadow: PropTypes.bool,
  imgUrl: PropTypes.string,
  alt: PropTypes.string,
  className: PropTypes.string,
  status: PropTypes.bool
}

Avatar.defaultProps = {
  shadow: false,
  alt: 'Profile photo',
  className: ''
}

export default Avatar
