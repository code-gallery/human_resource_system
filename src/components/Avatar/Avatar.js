import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Assets from 'components/AvatarBig/assets'
import './style.css'

const Avatar = (props) => {
  const { imageUrl, theme, size, label, icon } = props
  const image = (imageUrl && imageUrl.indexOf('https://') !== -1) ?
    imageUrl : Assets.placeholder
  const cssAvatar = classNames(`Avatar Avatar-${theme} Avatar-size-${size}`, {
    'Avatar-withLabel': label
  })
  const altText = (label) || 'logo'

  return (
    <div className={cssAvatar}>
      <img src={image} alt={altText} />
      {icon !== '' &&
        <span className={`Avatar-icon-${icon}`}>&nbsp;</span>
      }
      {label &&
        <span>{props.label}</span>
      }
    </div>
  )
}

Avatar.defaultProps = {
  theme: 'white',
  size: '40',
  icon: ''
}

Avatar.propTypes = {
  icon: PropTypes.string,
  label: PropTypes.string,
  theme: PropTypes.string,
  imageUrl: PropTypes.string,
  size: PropTypes.string
}

export default Avatar
