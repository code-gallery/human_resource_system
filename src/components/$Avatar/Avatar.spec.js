import React from 'react'
import renderer from 'react-test-renderer'
import Avatar from './index'

describe('<Avatar>', () => {
  const props = {
    imgUrl: 'https://example.com/assets/img/cat.jpg'
  }

  it('renders correctly', () => {
    const tree = renderer.create(
      <Avatar {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
