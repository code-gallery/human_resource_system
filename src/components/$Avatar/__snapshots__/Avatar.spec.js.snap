// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar> renders correctly 1`] = `
<img
  alt="Profile photo"
  className="$Avatar"
  src="https://example.com/assets/img/cat.jpg"
/>
`;
